src/ansys/api/sherlock/v0/SherlockLayerService.proto

```diff
@@ -566,4 +608,10 @@ service SherlockLayerService {
   
   // Copy one or more existing modeling regions.
   rpc copyModelingRegion(CopyModelingRegionRequest) returns (ReturnCode);
+
+  // List all the available layers as seen in the layer viewer for a project CCA.
+  rpc listLayers(ListLayersRequest) returns (ListLayersResponse);
+
+  // Export one or more 2D Layer Viewer images from a project CCA.
+  rpc exportLayerImage(ExportLayerImageRequest) returns (ExportLayerImageResponse);
```

Comment:
This should use server streaming. The client sends of list of requests. The server respondes for each element in the list. This would also mean you ExportLayerImageResponse does not have to have a repeated element, just a return code. This means Sherlock can return a return code per export layer image info in the ExportLayerImageRequest. One of the user inputs can fail without ruining all results.